### ISSUE TITLE
[Feature Request] NoisyBradleyTerry

### ISSUE THREAD

**Issue by shenxiangzhuang** [dustalov/evalica#13]
Hi, I'm wondering is there any plan to implement [NoisyBradleyTerry](https://crowd-kit.readthedocs.io/en/latest/pairwise/#crowdkit.aggregation.pairwise.NoisyBradleyTerry)? Which, I think can also be useful in practice.

**Comment by dustalov** [comment]
Hi, thank you for the request!

I have mixed feelings about bringing noisyBT, which is not a well-known method per se but a variation of another method, factorBT, published by our then-colleagues at Yandex ([Bugakova et al., 2019](https://arxiv.org/abs/1906.03711)). noisyBT is a simplified algorithm that does not take task features into account and only adds worker biases to the Bradley-Terry model.

However, even the simplified formulation makes it somewhat inconvenient to implement without a proper autograd. This is dangerous territory as one must either choose the preferred Python framework or implement everything from scratch. In [Crowd-Kit's implementation](https://github.com/Toloka/crowd-kit/blob/main/crowdkit/aggregation/pairwise/noisy_bt.py), we computed gradients and log-likelihood by hand, which impacted the maintainability (not even speaking about the readability) of the code.

Have you had any specific success stories with noisyBT?

**Comment by shenxiangzhuang** [comment]
Thanks for your timely reply!

I haven't read the paper of noisyBT/factorBT before, and it seems I underestimated the complexity of the algorithm. And I agree with your option that implement it from scratch maybe not a good idea while take the maintainability and readability into consideration.

I haven't any success story with noisyBT yet. To make it clear and avoid the XY Problem, my primary **requirement** is: **a robust pair to rank algorithm that works well when there are spammers' votes**. And I found noisyBT which might solve this problem. So I just create the issue to ask the reasons that why `evalica` don't include this algorithm.(Now I got it, I think~)

In short, I agree that include noisyBT in `evalica` is not necessary currently. And I would be grateful if you could give some advice on the spammers.(For now, I only learn about some anomaly detection process from [Chatbot Arena: An Open Platform for Evaluating LLMs by Human Preference](https://arxiv.org/pdf/2403.04132))

**Comment by dustalov** [comment]
Yes, it's a hugely complicated problem.

While noisyBT might tackle some positional bias of the annotators, I believe organizational means and statistical filters should be more effective than relying on a single relatively simple model of errors. For example, using IP-based features like Chatbot Arena does, using control tasks with known answers, or employing inter-annotator agreement information should cover more cases of malicious responses.

Note that Evalica supports assigning a weight to each response, similarly how it's done in Arena-Hard. At the same time, it is possible to encode our trust in this particular user as a weight (which needs to be estimated in advance).

**Comment by shenxiangzhuang** [comment]
Thanks a lot! And the suggestions are really helpful!
